csharp/test/Ice/retry/AllTests.cs

```diff
@@ -240,26 +241,26 @@ static public Test.IRetryPrx
                 }
                 catch (System.AggregateException ex)
                 {
-                    test(ex.InnerException is InvocationTimeoutException);
+                    test(ex.InnerException is TimeoutException);
                     Instrumentation.testRetryCount(2);
                     retry2.opIdempotent(-1); // Reset the counter
                     Instrumentation.testRetryCount(-1);
                 }
                 if (retry1.GetConnection() != null)
                 {
-                    // The timeout might occur on connection establishment or because of the sleep. What's
-                    // important here is to make sure there are 4 retries and that no calls succeed to
-                    // ensure retries with the old connection timeout semantics work.
+                    // The old invocationTimeout = -2 behaves like any timeout now, with no retry
+                    // TODO: should we remove this entirely? can the connection establishment timeout and trigger a
+                    // retry?
```

Comment:
I don't see the point of introducing new semantics for invocationTimeout = -2, I think we should just remove support for it.

Reply:
Done in latest commit